### PR TITLE
feat(maya): add destination address screen

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

Add an address input screen that handles the selected currency.

AddressInputFragment was moved to common to serve as a base class for Address Input Screens.
## How Has This Been Tested?

- [x] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [x] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: common/src/main/java/org/dash/wallet/common/payments/parsers/AddressParser.kt [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+package org.dash.wallet.common.payments.parsers
+
+import org.bitcoinj.core.Address
+import org.bitcoinj.core.Base58
+import org.bitcoinj.core.NetworkParameters
+
+open class AddressParser(pattern: String, val params: NetworkParameters?) {
+    companion object {
+        val PATTERN_BITCOIN_ADDRESS = "[${Base58.ALPHABET.joinToString(separator = "")}]{20,40}"
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/payments/parsers/AddressParser.kt:26]: This was moved from `wallet` but changed to a class that can handle different patterns.

File: common/src/main/java/org/dash/wallet/common/payments/parsers/AddressParser.kt [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                println(e)
+            }
+        }
+
+        return validRanges
+    }
+
+    protected open fun verifyAddress(addressCandidate: String) {
+        params?.let { Address.fromString(params, addressCandidate) }
+    }
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/payments/parsers/AddressParser.kt:72]: This is open so subclasses can verify the address in different ways besides what works for Dash.

File: common/src/main/java/org/dash/wallet/common/payments/parsers/Bech32.java [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+
+/**
+ * <p>Implementation of the Bech32 encoding.</p>
+ *
+ * <p>See <a href="https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki">BIP350</a> and
+ * <a href="https://github.com/bitcoin/bips/blob/master/bip-0173.mediawiki">BIP173</a> for details.</p>
+ */
+public class Bech32 {
+    /** The Bech32 character set for encoding. */
+    private static final String CHARSET = "qpzry9x8gf2tvdw0s3jn54khce6mua7l";
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/payments/parsers/Bech32.java:36]: this was taken from `bitcoinj` 0.16.x.  Newer bitcoin addresses use this format along with many other currencies.

File: common/src/main/java/org/dash/wallet/common/ui/scan/ScanActivity.java [hunk 4]
```diff
@@ -88,12 +59,38 @@
...
+import kotlin.Unit;
+
 /**
  * @author Andreas Schildbach
  */
 @SuppressWarnings("deprecation")
 @AndroidEntryPoint
-public final class ScanActivity extends LockScreenActivity
+public final class ScanActivity extends SecureActivity
         implements SurfaceTextureListener, ActivityCompat.OnRequestPermissionsResultCallback {
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/ui/scan/ScanActivity.java:94]: ScanActivity was translated into Kotlin in #1248

File: common/src/main/java/org/dash/wallet/common/data/PaymentIntent.java [hunk 5]
```diff
@@ -80,20 +84,27 @@ public boolean hasAmount() {
...
+        public String toString(NetworkParameters params) {
             final StringBuilder builder = new StringBuilder();
 
             builder.append(getClass().getSimpleName());
             builder.append('[');
             builder.append(hasAmount() ? amount.toPlainString() : "null");
             builder.append(',');
             if (ScriptPattern.isP2PKH(script) || ScriptPattern.isP2SH(script))
-                builder.append(script.getToAddress(Constants.NETWORK_PARAMETERS));
+                builder.append(script.getToAddress(params));
```
Comment (HashEngineering) [inline common/src/main/java/org/dash/wallet/common/data/PaymentIntent.java:101]: Remove dependency on the global `NETWORK_PARAMETERS`

**Review by Syn-McJ — APPROVED** [review]
Looks great

**Pull request merged**